Mathlib/Algebra/BigOperators/Finsupp/Basic.lean

```diff
@@ -543,6 +543,21 @@ theorem Finset.sum_apply' : (∑ k ∈ s, f k) i = ∑ k ∈ s, f k i :=
 theorem Finsupp.sum_apply' : g.sum k x = g.sum fun i b => k i b x :=
   Finset.sum_apply _ _ _
 
+theorem Finsupp.sum_apply'' {A F : Type*} [AddZeroClass A] [AddCommMonoid F] [FunLike F γ B]
+    (g : ι →₀ A) (k : ι → A → F) (x : γ)
+    (hg0 : ∀ (a : ι), k a 0 = 0) (hgadd : ∀ (a : ι) (b₁ b₂ : A), k a (b₁ + b₂) = k a b₁ + k a b₂)
+    (h0 : (0 : F) x = 0) (hadd : ∀ (f g : F), (f + g : F) x = f x + g x) :
+    g.sum k x = g.sum (fun i a ↦ k i a x) := by
+  induction g using Finsupp.induction with
```

Comment:
(Note: this can't easily be proven in terms of `Finsupp.sum_apply'` since the latter assumes `AddCommMonoid A` and this one only `AddZeroClass`.)